### PR TITLE
Fix for #850: Range Input order of attributes bug

### PR DESCRIPTION
**Pull request by mikewesthad**
Summary of issue:
When creating a range `input` by setting one attribute at a time, the order in which the attributes are set matters. The `value` attribute should be applied last, otherwise it may be clipped/rounded to match the browser default min/max/step values. See bug example in #850.

PR:
This PR fixes that by adjusting `diffAttributes` to delay setting value on `input` elements until all other attributes are handled. The associated test passes with this RP (and fails without it).

Background:
@gpoitch pointed out the relevant react code [here](https://github.com/facebook/react/blob/8ab56e5c8b8f797a6b2bdd4188b4082c55a5190b/src/renderers/dom/stack/client/wrappers/ReactDOMInput.js#L58-L71), which has similar logic for setting value last. (Thanks!)

### CODE REVIEW

**Comment by paranoidjk** [comment]
@MicheleBertoli typo. should be example in https://github.com/developit/preact/issues/850

File: src/vdom/diff.js [hunk 1]
```diff
@@ -293,6 +293,7 @@ export function removeChildren(node) {
  */
 function diffAttributes(dom, attrs, old) {
 	let name;
+	const isInput = dom.tagName.toLowerCase() === "input";
```
Comment (developit) [inline src/vdom/diff.js]: Need to use `dom.normalizedNodeName || dom.nodeName` here

Comment (mikewesthad) [reply src/vdom/diff.js]: Ah, thanks for taking a look. Realized `isNamedNode` is there for this exact purpose, so I dropped that in place of the line in question.

File: src/vdom/diff.js [hunk 2]
```diff
@@ -303,8 +304,14 @@ function diffAttributes(dom, attrs, old) {
...
 	for (name in attrs) {
-		if (name!=='children' && name!=='innerHTML' && (!(name in old) || attrs[name]!==(name==='value' || name==='checked' ? dom[name] : old[name]))) {
+		if (name==='children' || name==='innerHTML' || (isInput && name==='value')) continue;
+		if (!(name in old) || attrs[name]!==(name==='value' || name==='checked' ? dom[name] : old[name])) {
 			setAccessor(dom, name, old[name], old[name] = attrs[name], isSvgMode);
 		}
 	}
+
+	// update value last so that browser does not reject it for not fitting the default min/max/step
+	if (isInput && ("value" in attrs) && (!("value" in old) || attrs.value !== dom.value)) {
```
Comment (developit) [inline src/vdom/diff.js]: I bet we could just always special-case `value`, even for non-inputs.

**Comment by developit** [comment]
I'd like to get this into the next non-patch release, just need to golf it a bit to get the size hit down.

**Comment by mikewesthad** [comment]
Closing this now that this has been merged in elsewhere in #1435

**Pull request closed without merging**